lib/EarlyFunctionAnalysis/EarlyFunctionAnalysis.cpp

```diff
@@ -683,12 +748,13 @@ combineCrossCallSites(MetaAddress EntryPC, FunctionSummary &Summary) {
 }
 
 /// Perform cross-call site propagation
-static void interproceduralPropagation(const std::vector<CFEP> &Functions,
-                                       FunctionAnalysisResults &Properties) {
+static void
+interproceduralPropagation(FunctionAnalysisResults &Properties,
+                           SortedVector<model::Function> &Functions) {
   for (auto &J : Functions) {
```

Comment:
`J` is  bad name.

Reply:
I guess `K` (in the inner loop) is also a bad one. Do you have any suggestions by any chance? `First` and `Second` could sounds good?

Reply:
Now that I look at this piece of code it seems quite suspicious.
Why are you iterating over all the functions for each function?

Reply:
Well, if we need to cross-merge the results, this must be done for each call-site of a function (inner loop) for all the functions (outer loop). We keep fixed one, and in the meantime iterate over all the others (when the functions are the same we don't care).

Reply:
Uhmmm, I don't seem to be able to easily get the CallSites though, especially right about now that we are going to take the CFG out of the model?

Reply:
We're well before the model is needed here.

Can't you ask `FunctionAnalysisResults` to give all the functions it has?
Maybe, we don't need to pull in the model at all for this part.

Reply:
Separate commit for this.

Reply:
Uh? Is it fixed? If so, mark as resolved.